### PR TITLE
maybeError() calls callback synchronously

### PR DESCRIPTION
**Pull request by ralphtheninja**
Seems I was correct.

### CODE REVIEW

File: test/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+_levelup_test_db_61
```
Comment (ralphtheninja) [inline test/.gitignore]: Doh, did `echo _levelup_test_db* >> .gitignore`

**Comment by juliangruber** [comment]
ah, yes, but this won't work with the promises

**Comment by ralphtheninja** [comment]
> ah, yes, but this won't work with the promises

Why not?

**Comment by ralphtheninja** [comment]
@juliangruber I don't think promises work if we _don't_ do this. That is, the opposite of what your stating. But again, I might be wrong.

**Comment by vweevers** [comment]
And I think *both* would work 😄  Under the assumption that it doesn't matter when the callback is called; the promise will always resolve (to its consumer) in the next tick.

**Comment by ralphtheninja** [comment]
As long as I'm not messing up promises I'm happy. I can add a few similar tests using promises if you like.

**Comment by juliangruber** [comment]
I should have looked at the code! :D

It actually doesn't matter for promises, only for the regular callback case. With promises, you can resolve them before or after you return them.

+1 then!

**Pull request merged**